code/modules/food_and_drinks/food/snacks_frozen.dm

```diff
@@ -188,3 +191,73 @@
 	icon_state = "rainbow_sc"
 	list_reagents = list(/datum/reagent/consumable/nutriment = 5, /datum/reagent/consumable/laughter = 25)
 	tastes = list("ice" = 1, "water" = 1, "sunlight" = 5, "light" = 5, "slime" = 5, "paint" = 3, "clouds" = 3)
+
+/obj/item/reagent_containers/food/snacks/popsicle
+	name = "bug popsicle"
+	desc = "Mmmm, this should not exist."
+	icon = 'icons/obj/food/frozen_treats.dmi'
+	icon_state = "popsicle_stick_s"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+	tastes = list("beetlejuice")
+	trash = /obj/item/popsicle_stick
+	w_class = WEIGHT_CLASS_SMALL
+	var/overlay_state = "creamsicle_o" //This is the edible part of the popsicle.
+	var/bite_states = 4 //This value value is used for correctly setting the bitesize to ensure every bite changes the sprite. Do not set to zero.
+
+/obj/item/reagent_containers/food/snacks/popsicle/Initialize()
+	. = ..()
+	bitesize = reagents.total_volume / bite_states
+
+/obj/item/reagent_containers/food/snacks/popsicle/update_overlays()
+	. = ..()
+	if(bitecount)
+		. += "[initial(overlay_state)]_[min(bitecount, 3)]"
+	else
+		. += initial(overlay_state)
+
+/obj/item/reagent_containers/food/snacks/popsicle/On_Consume(mob/living/eater)
+	. = ..()
+	update_icon()
+
+/obj/item/popsicle_stick
+	name = "popsicle stick"
+	icon = 'icons/obj/food/frozen_treats.dmi'
+	icon_state = "popsicle_stick"
+	desc = "This humble little stick usually carries a frozen treat, at the moment it seems freed from this atlassian burden."
+	custom_materials = list(/datum/material/wood=20)
+	w_class = WEIGHT_CLASS_TINY
+	force = 0
+
+/obj/item/reagent_containers/food/snacks/popsicle/creamsicle_orange
+	name = "orange creamsicle"
+	desc = "A classic orange creamsicle. A sunny frozen treat."
+	list_reagents = list(/datum/reagent/consumable/orangejuice = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+
+/obj/item/reagent_containers/food/snacks/popsicle/creamsicle_berry
+	name = "berry creamsicle"
+	desc = "A vibrant berry creamsicle. A berry good frozen treat."
+	list_reagents = list(/datum/reagent/consumable/berryjuice = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+	overlay_state = "creamsicle_m"
+
+/obj/item/reagent_containers/food/snacks/popsicle/jumbo
+	name = "jumbo icecream"
+	desc = "A luxurius icecream covered in rich chocolate. It seems smaller than you remember it being."
+	list_reagents = list(/datum/reagent/consumable/hot_coco = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 3, /datum/reagent/consumable/sugar = 4)
+	overlay_state = "jumbo"
+
+/obj/item/reagent_containers/food/snacks/popsicle/nogga_black
```

Comment:
mate...
![](https://file.house/Oo3f.jpg)

Reply:
![image](https://user-images.githubusercontent.com/49783092/73120683-a0367800-3f71-11ea-82e4-11d86ad39749.png)

Nogga not nogger, two completely different words.

Reply:
mate...
![](https://file.house/Oo3f.jpg)

Reply:
haha funny the n word but its a different vowel hahaaaah

Reply:
they had to have known

Reply:
add gaytimes now